src/vma/dev/net_device_val.cpp

```diff
@@ -811,11 +811,16 @@ std::string net_device_val_eth::to_str()
 
 net_device_val_ib::~net_device_val_ib()
 {
-	g_p_neigh_table_mgr->unregister_observer(neigh_key(ip_address((in_addr_t)(inet_addr(BROADCAST_IP))), this), this);
+	struct in_addr in;
+	if (1 != inet_pton(AF_INET, BROADCAST_IP, &in)) {
```

Comment:
Aren't you looking for succeess return value? Shouldn't it be "equal to 1" rather than "not equal to 1"?
```code
if (1 == inet_pton(AF_INET, BROADCAST_IP, &in)) ...
```